server/lib/nl/common/utils.py

```diff
@@ -108,28 +108,40 @@ def sv_existence_for_places(places: List[str], svs: List[str],
 # keyed by SV DCID with value set to True if the SV has any
 # single data point series (across all places).
 def sv_existence_for_places_check_single_point(
-    places: List[str], svs: List[str],
+    places: List[str], svs: List[str], date: str,
     counters: ctr.Counters) -> Dict[str, bool]:
   if not svs:
     return {}, {}
 
   start = time.time()
   series_facet = fetch.series_facet(entities=places,
                                     variables=svs,
-                                    all_facets=False)
+                                    all_facets=bool(date))
   counters.timeit('sv_existence_for_places_check_single_point', start)
 
   existing_svs = {}
   existsv2places = {}
   for sv, sv_data in series_facet.get('data', {}).items():
     for pl, place_data in sv_data.items():
-      if not place_data.get('series'):
-        continue
-      num_series = place_data['series'][0]["value"]
-      existing_svs[sv] = existing_svs.get(sv, False) | (num_series == 1)
-      if sv not in existsv2places:
-        existsv2places[sv] = {}
-      existsv2places[sv][pl] = (num_series == 1)
+      for facet_data in place_data:
+        # If there is a date specified, but facet dates are different
+        # granularity, move on to next facet
```

Comment:
If the query asks for 2020, and we have 2020-06 I think we can answer?

Reply:
If you agree, have a TODO for revisiting (so we fix autopush first)?

Reply:
that makes sense to me, but will need more thought about how to pass this date to the chart config. Added a TODO & I will also create a buganizer issue for this.